### PR TITLE
Fix debug info for __local qualified variables

### PR DESCRIPTION
**Pull request by AlexeySotkin**
Current mechanism for searching of global variables via name lookup is not
reliable. With this patch we look through all global variables untill we find
one which has attached !dbg metadata, which refers via
DIGlobalVariableExpression to DIGlobalVariable which is being translated.

### CODE REVIEW

File: lib/SPIRV/LLVMToSPIRVDbgTran.cpp [hunk 1]
```diff
@@ -352,6 +352,18 @@ SPIRVEntry *LLVMToSPIRVDbgTran::getScope(DIScope *S) {
   }
 }
 
+SPIRVEntry *LLVMToSPIRVDbgTran::getGlobalVariable(const DIGlobalVariable *GV) {
+  for (llvm::GlobalVariable &V : M->globals()) {
```
Comment (svenvh) [inline lib/SPIRV/LLVMToSPIRVDbgTran.cpp]: You shouldn't need `llvm::` here.

File: lib/SPIRV/LLVMToSPIRVDbgTran.cpp [hunk 2]
```diff
@@ -352,6 +352,18 @@ SPIRVEntry *LLVMToSPIRVDbgTran::getScope(DIScope *S) {
...
+    SmallVector<DIGlobalVariableExpression *, 4> GVs;
+    V.getDebugInfo(GVs);
+    for (DIGlobalVariableExpression *GVE : GVs) {
+      if (GVE->getVariable() == GV)
+        return  SPIRVWriter->transValue(&V, nullptr);
```
Comment (svenvh) [inline lib/SPIRV/LLVMToSPIRVDbgTran.cpp]: nitpick: double space after `return`.

File: lib/SPIRV/LLVMToSPIRVDbgTran.cpp (hunk 1 above)
Comment (AlexeySotkin) [reply lib/SPIRV/LLVMToSPIRVDbgTran.cpp]: Fixed

File: lib/SPIRV/LLVMToSPIRVDbgTran.cpp (hunk 2 above)
Comment (AlexeySotkin) [reply lib/SPIRV/LLVMToSPIRVDbgTran.cpp]: Fixed

**Pull request merged**